### PR TITLE
docs: add "troubleshooting" for macOS

### LINKED ISSUE #20046: build failed on macOs Monterey 12.2.1
$ bazel build envoy
```
Starting local Bazel server and connecting to it...
INFO: Analyzed target //:envoy (742 packages loaded, 30490 targets configured).
INFO: Found 1 target...
INFO: From Executing genrule @io_opentracing_cpp//:generate_version_h:
-- The C compiler identification is AppleClang 12.0.5.12050022
-- The CXX compiler identification is AppleClang 12.0.5.12050022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- clang-tidy not found.
-- Configuring done
-- Generating done
-- Build files have been written to: /var/folders/lt/gstgm3fd0jj5hlcz0ryw6fxm0000gn/T/tmp.SIHmJjzC
-- Cache values
BUILD_DYNAMIC_LOADING:BOOL=ON
BUILD_MOCKTRACER:BOOL=OFF
BUILD_SHARED_LIBS:BOOL=ON
BUILD_STATIC_LIBS:BOOL=ON
BUILD_TESTING:BOOL=OFF
CLANG_TIDY_EXE:FILEPATH=CLANG_TIDY_EXE-NOTFOUND
CMAKE_BUILD_TYPE:STRING=
CMAKE_INSTALL_PREFIX:PATH=/usr/local
CMAKE_OSX_ARCHITECTURES:STRING=
CMAKE_OSX_DEPLOYMENT_TARGET:STRING=
CMAKE_OSX_SYSROOT:PATH=/Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk
ENABLE_LINTING:BOOL=ON
INFO: From Executing genrule @com_lightstep_tracer_cpp//:generate_version_h:
CMake Warning at CMakeLists.txt:74 (message):
  WITH_GRPC is not set; building without dynamic loading support.


-- The C compiler identification is AppleClang 12.0.5.12050022
-- The CXX compiler identification is AppleClang 12.0.5.12050022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /var/folders/lt/gstgm3fd0jj5hlcz0ryw6fxm0000gn/T/tmp.oaYAYREz
-- Cache values
BUILD_SHARED_LIBS:BOOL=ON
BUILD_STATIC_LIBS:BOOL=ON
CMAKE_BUILD_TYPE:STRING=
CMAKE_INSTALL_PREFIX:PATH=/usr/local
CMAKE_OSX_ARCHITECTURES:STRING=
CMAKE_OSX_DEPLOYMENT_TARGET:STRING=
CMAKE_OSX_SYSROOT:PATH=/Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk
DEFAULT_SSL_ROOTS_PEM:STRING=
HEADERS_ONLY:BOOL=ON
WITH_CARES:BOOL=OFF
WITH_DYNAMIC_LOAD:BOOL=ON
WITH_GRPC:BOOL=OFF
WITH_LIBEVENT:BOOL=OFF
ERROR: /private/var/tmp/_bazel_tuhao/12997281f702e09e8626091de078d39e/external/envoy/bazel/foreign_cc/BUILD:433:12: Foreign Cc - CMake: Building zlib failed: (Exit 1): bash failed: error executing command /bin/bash -c bazel-out/darwin-fastbuild/bin/external/envoy/bazel/foreign_cc/zlib_foreign_cc/wrapper_build_script.sh

Use --sandbox_debug to see verbose messages from the sandbox
rules_foreign_cc: Build failed!
rules_foreign_cc: Keeping temp build directory and dependencies directory for debug.
rules_foreign_cc: Please note that the directories inside a sandbox are still cleaned unless you specify --sandbox_debug Bazel command line flag.
rules_foreign_cc: Printing build logs:
_____ BEGIN BUILD LOGS _____

Bazel external C/C++ Rules. Building library zlib

xcrun: error: SDK "macosx12.1" cannot be located
xcrun: error: SDK "macosx12.1" cannot be located
xcrun: error: unable to lookup item 'Path' in SDK 'macosx12.1'
_____ END BUILD LOGS _____
rules_foreign_cc: Build wrapper script location: bazel-out/darwin-fastbuild/bin/external/envoy/bazel/foreign_cc/zlib_foreign_cc/wrapper_build_script.sh
rules_foreign_cc: Build script location: bazel-out/darwin-fastbuild/bin/external/envoy/bazel/foreign_cc/zlib_foreign_cc/build_script.sh
rules_foreign_cc: Build log location: bazel-out/darwin-fastbuild/bin/external/envoy/bazel/foreign_cc/zlib_foreign_cc/CMake.log

Target //source/exe:envoy-static failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 124.206s, Critical Path: 10.14s
INFO: 1757 processes: 1713 internal, 43 darwin-sandbox, 1 local.
FAILED: Build did NOT complete successfully
```

### PR DESCRIPTION
**Pull request by tuhao1020**
Signed-off-by: HaoTu haotu@alauda.io

Commit Message: add "trouble shooting" for macOS
Additional Description: 
Risk Level: 
Testing: 
Docs Changes: https://github.com/envoyproxy/envoy/blob/main/bazel/README.md
[Optional Fixes #Issue] Fixes #20046

### CODE REVIEW

File: bazel/README.md [hunk 1]
```diff
@@ -112,6 +112,28 @@ for how to update or override dependencies.
     The full version of Xcode (not just Command Line Tools) is also required to build Envoy on macOS.
     Envoy compiles and passes tests with the version of clang installed by Xcode 11.1:
     Apple clang version 11.0.0 (clang-1100.0.33.8).
+    
+    #### trouble shooting
```
Comment (adisuissa) [inline bazel/README.md:116]: `Troubleshoot` should be a single word. If the spell checker complains, you should add the word to the words database.

File: bazel/README.md [hunk 2]
```diff
@@ -112,6 +112,28 @@ for how to update or override dependencies.
...
+    ```
+    xcrun --show-sdk-version
+    ```
+    If the sdk version is lower than the one in the error message, upgrade your Command Line Tools using following commands:
+    ```
+    sudo rm -rf /Library/Developer/CommandLineTools
+    softwareupdate --all --install --force
+    sudo xcode-select --install
+    ```
+    If the following error occurs during the compilation process, please execute following command and retry.
```
Comment (adisuissa) [inline bazel/README.md:133]: What is the following error?

```suggestion
    If the following error occurs during the compilation process, please execute the following command and retry:
```

**Review by adisuissa — CHANGES_REQUESTED** [review]
Thanks!
Please also look at CI.
/wait

**Comment by rojkov** [comment]
The format check fails with the following suggestions:
```
##[error]: TESTS FAILED:
##[error]:  main@ /source/ci/format_pre.sh :47 (glint)
##[error]:      Please fix your editor to ensure:
##[error]:        - no trailing whitespace
##[error]:        - no mixed tabs/spaces
##[error]:        - all files end with a newline
```

**Comment by tuhao1020** [comment]
```
##[error]: trailing whitespace: bazel/README.md

##[error]: ERRORS found
##[error]: 0 files with missing newline
##[error]: 0 files with mixed tabs and spaces
##[error]: 1 files with trailing whitespace
```
It seems no line has trailing whitespace 😭  @rojkov @adisuissa

**Comment by rojkov** [comment]
Could you please merge the latest main branch to fix the tests?

**Comment by rojkov** [comment]
> Could you please merge the latest main branch to fix the tests?

Please disregard. No need to merge yet.

File: bazel/README.md [hunk 3]
```diff
@@ -113,6 +113,30 @@ for how to update or override dependencies.
...
+    ```
+
+    If the sdk version is lower than the one in the error message, upgrade your Command Line Tools using the following commands:
+    ```console
+    sudo rm -rf /Library/Developer/CommandLineTools
+    softwareupdate --all --install --force
+    sudo xcode-select --install
+    ```
+
+    If the following error occurs during the compilation process, please execute the following command and retry:
```
Comment (adisuissa) [inline bazel/README.md:135]: Can you explain what error is referred to here `If the following error occurs`?

**Review by adisuissa — COMMENTED** [review]
Looks better, thanks.
/wait

**Review by daixiang0 — APPROVED** [review]

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Review by rojkov — APPROVED** [review]
Thank you for resolving the issue! LGTM

**Pull request merged**